### PR TITLE
Default checking interval

### PR DESCRIPTION
**Pull request by benwhite-deltas**
closes hoodiehq/hoodie-connection-status#48

### CODE REVIEW

File: lib/start-checking.js [hunk 1]
```diff
@@ -6,12 +6,16 @@ internals.getOk = require('./get-ok')
 
 function startChecking (state, options) {
   options = parse(options)
-  handleInterval(state, options)
+  state !== undefined && state.method !== undefined && state.url !== undefined && handleInterval(state, options)
```
Comment (gr2m) [inline lib/start-checking.js]: `state` shouldn’t ever be undefined here, I think we can skip that check.

I’d do an early return here:

```js
if (!state.method || !state.url) {
  return
}

handleInterval(state, options)
```

File: lib/start-checking.js [hunk 2]
```diff
@@ -6,12 +6,16 @@ internals.getOk = require('./get-ok')
...
 }
 
 function handleInterval (state, options) {
   var timeout
   var ok = internals.getOk(state)
+  var timeoutHandler = function () {
```
Comment (gr2m) [inline lib/start-checking.js]: define as name function:

```js
function timeoutHandler () {
```

You can also define it outside

```js
function timeoutHandler (state, options) {
  var checkAgain = handleInterval.bind(null, state, options)
  internals.check(state, options).then(checkAgain, checkAgain)
}
```

and pass in state & options into [setTimeout](https://developer.mozilla.org/en-US/docs/Web/API/WindowTimers/setTimeout) like

```js
state.checkTimeout = setTimeout(timeoutHandler, timeout, state, options)
```

Both ways are good with me, what ever you think makes the code more readable :)

File: tests/unit/start-checking.js [hunk 3]
```diff
@@ -108,3 +108,36 @@ test('startChecking() with invalid options', function (t) {
...
+  t.is(startChecking.internals.check.lastCall.args[0], state, '1 requests sent')
+  clock.tick(29998)
+  t.is(startChecking.internals.check.callCount, 1, '1 requests sent')
+  clock.tick(1)
+  t.is(startChecking.internals.check.callCount, 2, '2 requests sent')
+
+  clearTimeout(state.checkTimeout)
+  simple.restore()
+  clock.uninstall()
+})
```
Comment (gr2m) [inline tests/unit/start-checking.js]: great test 👍

File: lib/start-checking.js (hunk 1 above)
Comment (benwhite-deltas) [inline lib/start-checking.js]: Agreed, changed to your suggestion

File: lib/start-checking.js (hunk 2 above)
Comment (benwhite-deltas) [inline lib/start-checking.js]: Now defined in module scope as suggested as a name function; is there any style guide that says that name functions are preferred, and if not, would it be good to add one?

File: tests/unit/start-checking.js (hunk 3 above)
Comment (benwhite-deltas) [inline tests/unit/start-checking.js]: Thanks!

**Comment by gr2m** [comment]
Thanks! Sorry it took us so long to get it merged & released

**Pull request merged**